Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -2025,172 +2120,203 @@ private void UpdateRigHandles()
+                rigRoot.position = TargetBounds.bounds.center;
                 rigRoot.rotation = Target.transform.rotation;
                 rigRoot.parent = transform;
             }
         }
 
-        private HandleProximityState ScaleHandleByProximity(Transform handle, Renderer renderer, Transform handleVisual, HandleProximityState state, List<Vector3> points, float handleSize)
+        private void HandleProximityScaling()
         {
-            HandleProximityState newState = state;
-            float closestDistance = float.MaxValue;
-            foreach (Vector3 point in points)
-            {
-                closestDistance = Mathf.Min(closestDistance, (handle.position - point).magnitude);
-            }
-
-            if (closestDistance < handleCloseProximity)
-            {
-                newState = HandleProximityState.CloseProximity;
-            }
-            else if (closestDistance < handleMediumProximity)
-            {
-                newState = HandleProximityState.MediumProximity;
-            }
-            else//far
+            //only use proximity effect if nothing is being dragged or grabbed
+            if (currentPointer == null)
             {
-                newState = HandleProximityState.FullsizeNoProximity;
-            }
-
-            ScaleHandle(handle, handleVisual, newState, handleSize);
-
-            if (renderer != null && state != newState)
-            {
-                renderer.material = newState == HandleProximityState.CloseProximity ? handleGrabbedMaterial : handleMaterial;
-            }
-            return newState;
-        }
-
-        private void ScaleHandle(Transform handleRoot, Transform handleVisual, HandleProximityState targetState, float handleSize)
-        {
-            float targetScale = targetState == HandleProximityState.CloseProximity ? closeScale : (targetState == HandleProximityState.MediumProximity ? mediumScale : farScale);
-            float weight = targetState == HandleProximityState.CloseProximity ? closeGrowRate : (targetState == HandleProximityState.MediumProximity ? mediumGrowRate : farGrowRate);
-            float newLocalScale = (handleVisual.transform.localScale.x * (1.0f - weight)) + (handleSize * targetScale * weight);
-            handleVisual.transform.localScale = new Vector3(newLocalScale, newLocalScale, newLocalScale);
-        }
+                proximityPointers.Clear();
+                proximityPoints.Clear();
 
-        private void HandleProximityScaling(bool forceFar = false)
-        {
-            if (corners?.Count > 0 || balls?.Count > 0)
-            {
-                inputPoints.Clear();
-                //only use proximity effect if nothing is being dragged or grabbed
-                if (currentPointer == null && forceFar == false)
+                // Find all valid pointers
+                foreach (var inputSource in CoreServices.InputSystem.DetectedInputSources)
```

Comment:
Not needed for this PR, but I feel like we should create an IEnumerator in InputSystem that enumerates over all pointers, since I imagine many people want to iterate over all pointers in the scene.

Reply:
Love it!